clang/lib/CodeGen/CGLoopInfo.h

```diff
@@ -82,6 +82,12 @@ struct LoopAttributes {
   /// Value for 'llvm.loop.align' metadata.
   unsigned CodeAlign;
 
+  /// Argument values of llvm.loop.itercount metadata;
+  struct {
+    std::optional<int64_t> Lwb;
+    std::optional<int64_t> Upb;
```

Comment:
Just a silly question from my side: why we care about Upb if take the information after from Lwb?

Reply:
We are trying to implement LOOP_RANGE, which has an upper bound. Later we may not bother to create a three stage pipeline if max_iter == 2. I don't know.